### PR TITLE
remove code related to encrypted file download button in iframe

### PR DESCRIPTION
**Pull request by kerryarchibald**
Fixes: https://github.com/vector-im/element-web/issues/6418

Removes unused code

---
This change is marked as an *internal change* (Task), so will not be included in the changelog.

Preview: https://pr7940--matrix-react-sdk.netlify.app
⚠️ Do you trust the author of this PR? Maybe this build will steal your keys or give you malware. Exercise caution. Use test accounts.

### CODE REVIEW

File: src/components/views/messages/MFileBody.tsx [hunk 1]
```diff
@@ -274,20 +230,6 @@ export default class MFileBody extends React.Component<IProps, IState> {
...
-                            future, the download link may disappear entirely at which point it could also
-                            be suitable to just remove this bit of code.
-                         */ }
-                        <iframe
-                            aria-hidden
-                            title={presentableTextForFile(this.content, _t("Attachment"), true, true)}
-                            src={url}
-                            onLoad={() => this.downloadFile(this.fileName, this.linkText)}
-                            ref={this.iframe}
-                            sandbox="allow-scripts allow-downloads allow-downloads-without-user-activation" />
```
Comment (t3chguy) [inline src/components/views/messages/MFileBody.tsx:290]: Pretty sure this iframe is still needed, this.iframe is passed into this.fileDownloader's c'tor

File: src/components/views/messages/MFileBody.tsx [hunk 2]
```diff
@@ -143,17 +107,11 @@ export default class MFileBody extends React.Component<IProps, IState> {
         return presentableTextForFile(this.content);
     }
 
-    private downloadFile(fileName: string, text: string) {
+    private downloadFile(fileName: string) {
         this.fileDownloader.download({
```
Comment (t3chguy) [inline src/components/views/messages/MFileBody.tsx:111]: calling `.download` requires the iframe. See FileDownloader::download and the postMessage it needs to send.

**Review by t3chguy — APPROVED** [review]
Looks sane!

File: src/components/views/messages/MFileBody.tsx (hunk 1 above)
Comment (kerryarchibald) [reply src/components/views/messages/MFileBody.tsx:290]: As discussed in chat, FileDownloader creates an equivalent iframe when none is passed to the constructer. Removed it from the ctor here, and the ref.

**Pull request merged**